### PR TITLE
Include file path in errors from PartialDriver.ReadUp() and ReadDown()

### PR DESCRIPTION
**Pull request by leventov**
Helps to diagnose errors like https://github.com/golang-migrate/migrate/issues/79

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -62,4 +62,4 @@ require (
 	modernc.org/zappy v1.0.0 // indirect
 )
 
-go 1.13
+go 1.14
```
Comment (dhui) [inline go.mod:65]: revert this change since we still support go 1.13
https://golang.org/ref/mod#tmp_12

File: source/httpfs/partial_driver.go [hunk 2]
```diff
@@ -108,9 +108,14 @@ func (p *PartialDriver) Next(version uint) (nextVersion uint, err error) {
 // ReadUp is part of source.Driver interface implementation.
 func (p *PartialDriver) ReadUp(version uint) (r io.ReadCloser, identifier string, err error) {
 	if m, ok := p.migrations.Up(version); ok {
-		body, err := p.fs.Open(path.Join(p.path, m.Raw))
+		upPath := path.Join(p.path, m.Raw)
+		body, err := p.fs.Open(upPath)
 		if err != nil {
-			return nil, "", err
+			return nil, "", &os.PathError{
```
Comment (dhui) [inline source/httpfs/partial_driver.go:114]: I'm wary of unnecessarily wrapping errors.
Do you have specific examples of `http.Filesystem` implementations that don't return an `os.PathError` when `Open()` fails?

The [issue](https://github.com/golang-migrate/migrate/issues/79) you've referenced existed before we added `PartialDriver`.

Comment (leventov) [reply source/httpfs/partial_driver.go]: > Do you have specific examples of http.Filesystem implementations that don't return an os.PathError when Open() fails?

I actually didn't check that before you asked but it turns out that yes, for example, `_escStaticFS` can return bare `os.ErrNotExist`.

File: go.mod (hunk 1 above)
Comment (leventov) [reply go.mod]: Reverted

File: source/httpfs/partial_driver.go [hunk 3]
```diff
@@ -136,3 +137,20 @@ func (p *PartialDriver) ReadDown(version uint) (r io.ReadCloser, identifier stri
...
 }
+
+func (p *PartialDriver) open(path string) (http.File, error) {
+	f, err := p.fs.Open(path)
+	if err == nil {
+		return f, nil
+	}
+	// Some non-standard file systems may return errors that don't include the path, that
+	// makes debugging harder.
+	if !errors.As(err, new(*os.PathError)) {
```
Comment (dhui) [inline source/httpfs/partial_driver.go:148]: Use `errors.Is()` if the target is unused.

Comment (leventov) [reply source/httpfs/partial_driver.go:148]: `errors.Is()` has different semantic - exact value match. I need just type match.

**Comment by leventov** [comment]
@dhui Fixed. Build failure now seems because of docker container pull problems only.
